sys-devel/gcc/gcc-11.2.0_2021_07_28.recipe

```diff
@@ -261,13 +266,57 @@ BUILD()
 		"../config.h"
 	sed -i -e 's,#define _GLIBCXX_HAVE_TLS 1,/* #undef _GLIBCXX_HAVE_TLS */,' \
 		"../include/$effectiveTargetMachineTriple/bits/c++config.h"
-	make CFLAGS="-O2 $kernelCcFlags" CXXFLAGS="-O2 $kernelCcFlags"
+	make $jobArgs CFLAGS="-O2 $kernelCcFlags" CXXFLAGS="-O2 $kernelCcFlags"
 	mv .libs/libsupc++.a .libs/libsupc++-kernel.a
 	mv saved/libsupc++* .libs/
 	mv saved/gthr-default.h "../include/$effectiveTargetMachineTriple/bits/"
 	mv saved/config.h ..
 	mv saved/c++config.h "../include/$effectiveTargetMachineTriple/bits/"
 	rmdir saved
+
+	local bootCcFlags
+	if [ $effectiveTargetArchitecture == arm ]; then
+		# EFI arm (32-bit) requires software fp
+		bootCcFlags+="-mfloat-abi=soft -nostartfiles -fshort-wchar";
+	fi
+
+	echo "### libgcc-boot ###"
+
+	# build bootloader version of libgcc and libgcc_eh.a (no threads, TLS)
+	cd $objectsDir/$effectiveTargetMachineTriple/libgcc
+	mkdir -p saved
+	mv libgcc.a libgcc_eh.a libgcc_s* libgcov* libgcc-kernel.a libgcc_eh-kernel.a saved/
+	make clean
+	ln -sfn "$sourceDir/libgcc/gthr-single.h" gthr-default.h
+	make $jobArgs CFLAGS="-O2 $bootCcFlags" CXXFLAGS="-O2 $bootCcFlags"
+	mv libgcc.a libgcc-boot.a
+	mv libgcc_eh.a libgcc_eh-boot.a
+	ln -sfn "$sourceDir/libgcc/gthr-posix.h" gthr-default.h
+	mv saved/* .
+	rmdir saved
+
+	echo "### libsupc++-boot ###"
+
+	# build bootloader version of libsupc++.a (without threads or TLS)
+	cd $objectsDir/$effectiveTargetMachineTriple/libstdc++-v3/libsupc++
+	mkdir -p saved
+	mv .libs/libsupc++* saved/
+	cp "../include/$effectiveTargetMachineTriple/bits/gthr-default.h" \
+		"../config.h" \
+		"../include/$effectiveTargetMachineTriple/bits/c++config.h" \
+		saved/
+	make clean
+	# deactivate threads and tls support
+	cp "../include/$effectiveTargetMachineTriple/bits/gthr-single.h" \
+		"../include/$effectiveTargetMachineTriple/bits/gthr-default.h"
+	sed -i -e 's,#define _GLIBCXX_HAS_GTHREADS 1,/* #undef _GLIBCXX_HAS_GTHREADS */,' \
+		"../config.h"
+	sed -i -e 's,#define _GLIBCXX_HAVE_TLS 1,/* #undef _GLIBCXX_HAVE_TLS */,' \
+		"../include/$effectiveTargetMachineTriple/bits/c++config.h"
+	make $jobArgs CFLAGS="-O2 $bootCcFlags" CXXFLAGS="-O2 $bootCcFlags"
+	mv .libs/libsupc++.a .libs/libsupc++-boot.a
+	mv saved/* .libs/
```

Comment:
this is incorrect, please see how it is done for kernel.

Reply:
Yes, I can see. We have to be more careful when "restoring" the original files after libsupc++-boot.a has been built.
I'll try to rework - and surely I'll do more in-depth testing before submitting it for review.